### PR TITLE
Move autorest.go code generator to packages subdirectory

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
This makes the layout consistent with other packages and sets us up for future addition of a TypeSpec package.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
@benbp I renamed ci.yml to ci-go.yml to better distinguish the CI files which of course hosed part of CI.  Is there a way I can modify the pipeline so this new one is picked up?

File: packages/autorest.go/tsconfig.json
Comment (RickWinter) [inline packages/autorest.go/tsconfig.json:1]: Are these changes b/c we are using a more recent node?

File: packages/autorest.go/test/synapse/2019-06-01/azspark/zz_models.go
Comment (RickWinter) [inline packages/autorest.go/test/synapse/2019-06-01/azspark/zz_models.go:1]: Did you intend for this cleanup?

Comment (jhendrixMSFT) [reply packages/autorest.go/test/synapse/2019-06-01/azspark/zz_models.go:1]: There was a recent change in the autorest front-end that introduced this delta.  Not sure why and I will look into that, but for now, the change is expected.

**Review by RickWinter — APPROVED** [review]

File: packages/autorest.go/tsconfig.json
Comment (jhendrixMSFT) [reply packages/autorest.go/tsconfig.json:1]: No, I'm just making this current.  I "borrowed" it from the autorest.typescript tsconfig.

Comment (jhendrixMSFT) [reply packages/autorest.go/tsconfig.json:1]: In addition, the majority of these settings were in the root tsconfig.json which has been removed.  I did find two settings I missed though and have added them back.

**Pull request merged**